### ISSUE TITLE
util: test_timeline - show amount of time between last matched line and the end of the encounter

### ISSUE THREAD

**Issue by xiashtra** [quisquous/cactbot#5617]
When testing older timelines to make sure they still work properly, sometimes it's not immediately apparent when the timeline drastically desyncs or the encounter continues past the end of the timeline.  It would be nice if the `test_timeline` utility indicated the delta between the last matched timeline entry and the last line of an encounter, to make it more obvious when this happens.  This could possibly also be worked into a heuristic to warn when there is a large window between timeline syncs mid-encounter.

**Comment by quisquous** [comment]
Yeah, I think test_timeline should probably take note of when combat finishes or the log is done and then throw errors for everything it maybe should have hit if the timeline kept running for X seconds?

Yeah also agree that if we get to a point where syncs have been missed and we've gone for X seconds, it should be more obvious.  (Maybe in debug mode the timeline gets a red outline or something?)
